src/services/content.ts

```diff
@@ -260,6 +262,30 @@ const handleBatchAnnouncement = (dispatch: Dispatch) => (
   dsnp.readBatchFile(batchAnnouncement, (announcementRow, batchIndex) => {
     try {
       const announcement = announcementRow as unknown;
+
+      if (!isSignedAnnouncement(announcement)) {
+        console.log(
+          "batched announcement is not signed announcment",
+          announcement
+        );
+        return;
+      }
+
+      if (
+        !core.contracts.registry.isSignatureAuthorizedTo(
+          announcement.signature,
+          announcement,
+          BigInt(announcement.fromId),
+          Permission.ANNOUNCE
+        )
+      ) {
+        console.log(
+          "batched announcment has unauthorized signature",
+          announcement
+        );
+        return;
```

Comment:
Do we want to show invalid stuff, but flag it? I guess this is a good first piece either way.

Reply:
I don't think so. At best it would be vector to pollute the feed. At worst it would leave doubt about the authenticity of counterfeit content. There's also no real way to do it for profiles or graph changes without allowing malicious behavior.